software/qualimap/bamqc/main.nf

```diff
@@ -0,0 +1,76 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+def options    = initOptions(params.options)
+
+process QUALIMAP_BAMQC {
+    tag "$meta.id"
+    label 'process_medium'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:meta.id) }
+
+    conda (params.enable_conda ? "bioconda::qualimap=2.2.2d" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/qualimap:2.2.2d--1"
+    } else {
+        container "quay.io/biocontainers/qualimap:2.2.2d--1"
+    }
+
+    input:
+    tuple val(meta), path(bam)
+    path regions
+
+    output:
+    tuple val(meta), path("${prefix}"), emit: results
+    path  "*.version.txt"             , emit: version
+
+    script:
+    def software   = getSoftwareName(task.process)
+    prefix         = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+
+    def memory = 3
+    if (!task.memory) {
+        log.info '[Qualimap BamQC] Available memory not known - defaulting to 3GB. Specify process memory requirements to change this.'
+    } else {
+        memory = task.memory.toGiga()
+    }
+    def threads    = task.cpus
+    def collect_pairs = meta.single_end ? '' : '--collect-overlap-pairs'
+
+    // currently there is no convenient way to have optional input in dsl2
+    // there needs to be a placeholder file in the workflow definition with the magic name 'dummy_file.txt'
+    def regions_file = regions.name != 'dummy_file.txt' ? '--gff ${regions}' : ''
+
+    def gcref = ''
+    if (meta.genome.toString().startsWith('GRCh')) {
+        gcref = '-gd HUMAN'
+    } else if (meta.genome.toString().startsWith('GRCm')) {
+        gcref = '-gd MOUSE'
+    }
```

Comment:
This should be evaluated and added to `options.args` within the pipeline and not the module. e.g. [here](https://github.com/nf-core/rnaseq/blob/3643a94411b65f42bce5357c5015603099556ad9/rnaseq.nf#L122)